basil.js

```diff
@@ -1844,14 +1844,25 @@ pub.previousPage = function () {
 
 
 /**
- * The number of all pages in the current document.
+ * Returns the number of all pages in the current document. If a number is given as an argument,
+ * it will set the document's page count to the given number by either adding pages or removing
+ * pages until the number is reached. If pages are added, the master page of document's last
+ * page will be applied to the new pages.
  *
  * @cat Document
  * @subcat Page
  * @method pageCount
+ * @param  {Number} [pageCount] New page count of the document (integer between 1 and 9999).
  * @return {Number} The amount of pages.
  */
-pub.pageCount = function() {
+pub.pageCount = function(pageCount) {
+  if(arguments.length) {
+    if(pageCount.constructor.name === "Number" && pageCount % 1 === 0 && pageCount > 0 && pageCount < 10000) {
```

Comment:
Also wouldn't it be more readable if it was a bit more like this:  

```js
if(arguments.length > 0) {
	if(pageCount.constructor.name === "Number"){
		if(pageCount % 1 === 0){
			if (pageCount > 0 && pageCount < 10000) {
				
			}
		}
	}
}
```

I think this is mostly a question of style. I always dislike cramming to many conditions into one `()`

Reply:
Hm, I personally find 4 nested if statements to be less readable than 4 chained ones. Yes, it's not ideal to have to many conditions in one `()`, but nesting 4 levels deep is not ideal either.

We could maybe add an internal helper function `isInteger()`. Then it would be a bit more easy to read and we don't need to put a comment somewhere to make it even more messy.

Then it would read
```js
if(isInteger(pageCount) && pageCount > 0 && pageCount < 10000) {

}
```

Reply:
Yeah. Maybe that’s a way. 4 nested conditions is to much. That’s right. I just think it should be more expressive to get the gist.

Reply:
Sure thing. I'll add `isInteger()`. If we have it anyways, should we also expose it to the user? We have `isNumber()`, `isString()` and `isArray()` already, so I guess there would be no harm in adding another one. What do you think?

Reply:
Sure. Why not.